### DISCUSSION TITLE
Return two result sets from a custom controller or a join

### DISCUSSION THREAD

**Discussion by basavarajdodamani** [strapi/strapi#8006] [Legacy Topics]
Hello,
Can we return two result sets from a custom controller which fetches result sets from two different collections in **one invocation**?
![image](https://user-images.githubusercontent.com/69140968/94014268-eb3ab080-fdc8-11ea-85cb-6fcc43aaf946.png)

or

I would like to get a result set from multiple collections based on a common column. Just like returning result set based on join on common field in RDBMS. (**again in one invocation**)

Select c1.a, c1,b, c2.c, c2.d, c3.e, c3.f, c4.g, c4.h from c1,c2,c3,c4 on c1.id=c2.id and c3.id=c4.id?

Datastore: MONGODB

Kind Regards,
Basu

**Comment by derrickmehaffy** [comment]
Sure, you just need to set two variables and perform the service queries then construct your response object

```js
const { sanitizeEntity } = require('strapi-utils');

module.exports = {
  /**
   * Retrieve records from two collections.
   *
   * @return {Array}
   */

  async findMultiple(ctx) {
    let collectionA;
    let collectionB;
    if (ctx.query._q) {
      collectionA = await strapi.services.collectionA.search(ctx.query);
      collectionB = await strapi.services.collectionB.search(ctx.query);
    } else {
      collectionA = await strapi.services.collectionA.find(ctx.query);
      collectionB = await strapi.services.collectionB.find(ctx.query);
    }

    return {
      collectionA: entities.map(entity => sanitizeEntity(entity, { model: strapi.models.collectionA })),
      collectionB: entities.map(entity => sanitizeEntity(entity, { model: strapi.models.collectionB }))
    }
  },
};
```

Adjust as needed, that is just a very basic example but so long as you are returning json you can structure the response as you need to.

**Reply by basavarajdodamani** [reply]
thanks @derrickmehaffy for the quick response. This works fine. 
how could i achieve something like below:
Select c1.a, c1,b, c2.c, c2.d, c3.e, c3.f, c4.g, c4.h from c1,c2,c3,c4 on c1.id=c2.id and c3.id=c4.id?

This is basically, one array having fields from different collections c1,c2,c3,c4 based on some matching criteria.

**Reply by derrickmehaffy** [reply]
For something like that I think you would need to use something like custom queries directly hooking into mongoose: https://strapi.io/documentation/v3.x/concepts/queries.html#mongoose

If you were using a SQL database I would say you could use the `strapi.connections.default.raw('Some SQL Join statement here')` but I'm unsure on mongodb
